torchgen/native_function_generation.py

```diff
@@ -405,14 +406,14 @@ def add_generated_native_functions(
             has_view_ops = any(
                 f.is_view_op and str(f.func.name.name) != "set_" for f in d.values()
             )
-            # Don't generate the other variants for CompositeImplicitAutograd operators.
+            # Don't generate the other variants for non-core CompositeImplicitAutograd operators.
             # We could probably do this, but the main benefit of generating the function triplets
             # is for transforms that need them, and transforms don't need to act directly
             # on CompositeImplicitAutograd operators (since we let them decompose).
             are_composite_implicit = all(
                 f.has_composite_implicit_autograd_kernel for f in d.values()
             )
-            if are_manual or has_view_ops or are_composite_implicit:
+            if are_manual or has_view_ops or are_composite_implicit and not is_core:
```

Comment:
hmm @GregoryComer one invariant I think we want to keep is that for **any** ATen ops that we auto-generate here, they have an `autogen: <op_name>` entry in `native_functions.yaml` (e.g. here: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L12610)

It looks like that empirically isn't completely true today... given that I see `autogen: upsample_bilinear2d.vec_out` in the yaml, but `torch.ops.aten.upsample_bilinear2d.vec_out` doesn't exist (@larryliu0820 does this ring a bell for you?)

But at the very least, we can keep the invariant in the other direction: instead of **always** doing this for ops marked as "core", can we also require them to exist in the yaml and have the `autogen` tag?

Reply:
Thanks. I have a separate PR out to allow autogen of core CIA ops - right now, autogen is skipped for all CIA ops. Any concerns with this approach: https://github.com/pytorch/pytorch/pull/141797?

Reply:
FYI This PR (https://github.com/pytorch/pytorch/pull/1417970) is included here. I was hitting issues with ghstack due to the cross-repo dep, but I can rebase if/once that lands.

Regarding this check, I see there is a needs_out check already in place that checks to see if there is an autogen variant. However, it is getting skipped for CIA ops. I could just remove the are_composite_implicit from above, if we want to do that.

Reply:
@bdhirsh We will continue/skip if a native function has a CIA kernel in line 417, and only check the `autogen` tag in line 471. So basically "having CIA kernel" overrides "having `autogen` tag"